src/medCore/gui/viewContainers/medViewContainer.cpp

```diff
@@ -829,3 +859,115 @@ void medViewContainer::removeColorIndicator(QColor color)
 {
     d->poolIndicator->removeColorIndicator(color);
 }
+
+/**
+ * @brief saves the scene in a XML file
+ * Saves views (all layers), and toolboxes parameters
+ * 	expected tree is as follow:
+ *  workingDir (user-defined)
+ * 		|-viewID0
+ * 			|-mapping.xml
+ * 			|-layerID01.xml
+ * 			|-layerID02.xml
+ * 			|-data_file (as saved by the dedicated writer)
+ */
+QString medViewContainer::saveScene()
+{
+    QString dirPath = QFileDialog::getExistingDirectory(this, tr("Open Directory"),"/home",QFileDialog::ShowDirsOnly);
+    QDir workingDir(dirPath);
+    QDomDocument doc("xml");
+    QDomElement root=doc.createElement("scene");
+    doc.appendChild(root);
+
+    QString subDirName;
+
+    medAbstractLayeredView* layeredView=dynamic_cast<medAbstractLayeredView*>(view());
+    if (layeredView!=0)
+    {
+        subDirName="view"+QUuid::createUuid ().toString();
+        if(!workingDir.mkdir(subDirName))
+            qWarning() << " failed to create new directory: "<<subDirName;
+        workingDir.cd(subDirName);
+        QString generatedPath=workingDir.canonicalPath()+"/mapping.xml";
+        QDomElement layeredViewInfo=doc.createElement("layeredView");
+        layeredViewInfo.setAttribute("path",subDirName+"/mapping.xml");
+        layeredViewInfo.setAttribute("id",layeredView->identifier());
+        root.appendChild(layeredViewInfo);
+        layeredView->write(generatedPath);
+        workingDir.cdUp();
+    }
```

Comment:
Technically it's not a pb, it will create an almost empty file that won't do much.
The real question is more: is there a way the cast doesn't work? For now I have no idea.
What do you think?